gp_services.R

```diff
@@ -40,13 +40,38 @@ service_typesnew1 <- gsub("Esri REST Tile Service", "agsmapserver", service_type
 service_typesnew2 <- gsub("Esri REST Feature Service", "agsfeatureserver", service_typesnew1)
 service_typesnew3 <- gsub("Esri REST Image Service", "agsimageserver", service_typesnew2)
 service_typesnew4 <- gsub("OGC Web Map Service \\(WMS\\)", "wms", service_typesnew3)
-service_typesnew5 <- gsub("OGC Web Feature Service \\(WFS\\)", "wfs", service_typesnew4)
+serviceType <- gsub("OGC Web Feature Service \\(WFS\\)", "wfs", service_typesnew4)
 
 #Create a data frame of the info so far
-all_service_info <- data.frame(service_ids, service_label, service_typesnew5, service_urls)
+all_service_info <- data.frame(id, title, serviceType, serviceUrl, stringsAsFactors = FALSE)
 View(all_service_info) #View the data frame
 
 #Write the data frame to a CSV
 write.table(all_service_info, quote=FALSE, row.names = FALSE, col.names = c("id", "title", "serviceType", "serviceUrl"), file = "all_service_info.csv", sep = ",")

```

Comment:
We could break this into multiple lines to enhance readability:

```
write.table(
    all_service_info,
    quote=FALSE,
    row.names = FALSE,
    col.names = c("id", "title", "serviceType", "serviceUrl"),
    file = "all_service_info.csv",
    sep = ",
)
```

The filename could be put in a constant.